### ISSUE TITLE
[Bug]: State "unknown"

### ISSUE THREAD

**Issue by Maddog8902** [mampfes/hacs_waste_collection_schedule#2035]
### I Have A Problem With:

The integration in general

### What's Your Problem

Since 5 days the Integration shows in every sensor that the state ist "unknown".
It works for a few months and didnt changed anything on the integration.

### Source (if relevant)

de.k4systems.abfallappmyk

### Logs

```Shell
2024-04-29 17:26:50.344 ERROR (SyncWorker_1) [waste_collection_schedule.source_shell] fetch failed for source Apps by Abfall+:
Traceback (most recent call last):
File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
entries = self._source.fetch()
^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/app_abfallplus_de.py", line 140, in fetch
for d in self._app.generate_calendar():
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py", line 786, in generate_calendar
self.select_kommune()
File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py", line 502, in select_kommune
raise Exception(f"Region {self._region_search} not found.")
Exception: Region Mendig not found.
```

### Relevant Configuration

```YAML
# MuellKalender
waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.abfallappmyk
        city: Mendig
        strasse: Alle Straßen
      customize:
       - type: Restabfall
         alias: rest
         icon: mdi:trash-can
       - type: Gelber Sack
         alias: gelbersack
         icon: mdi:recycle
       - type: Bioabfall
         alias: bio
         icon: mdi:bio
       - type: Papierabfall
         alias: papier
         icon: mdi:PackageVariant
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [ ] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by coding-lemur** [comment]
I have the same issue. No waste collection entries in the HA calendar anymore :/

**Comment by chrizzo84** [comment]
Wait for v1.48 or just take master as specific version.
This bug is fixed and merged to master but not officially released yet.

**Comment by 5ila5** [comment]
Duplicate of #1989

**Comment by voku** [comment]
@chrizzo84 How can I use the master branch via HACS?

**Comment by 5ila5** [comment]
The fix will be part of release 1.48.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)

I'll probably release 1.48.0 very soon as app_abfallplus_de seems to be by quite a lot of people
